src/serialize.h

```diff
@@ -67,54 +67,54 @@ extern "C" {
 #define LAST_TAG 57
 
 #define write_uint8(s, n) ios_putc((n), (s))
-#define read_uint8(s) ((uint8_t)ios_getc(s))
-#define write_int8(s, n) write_uint8(s, n)
-#define read_int8(s) read_uint8(s)
+#define read_uint8(s) ((uint8_t)ios_getc((s)))
+#define write_int8(s, n) write_uint8((s), (n))
+#define read_int8(s) read_uint8((s))
```

Comment:
How about instead turning these also into `static inline` wrappers? And also `write_uint8`. That solves the same problem, and is consistent with the rest of the file.